Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm

```diff
@@ -207,7 +210,12 @@
         return false;
     }
 
+    uint64_t seekTo = std::exchange(m_seekTo, NoSeek);
```

Comment:
push and pull may be done in different threads, we probably want to have `m_seekTo` as an atomic.

Reply:
> push and pull may be done in different threads, we probably want to have `m_seekTo` as an atomic.

The code as it is existing isn't thread-safe. We have several non-atomic member variables being accessed in both the pull and push method: `m_waitToStartForPushCount` , `m_muted` or even the `m_scratchBuffer` ; if push and pull were concurrent there would be serious trouble.